src/Certify.Core/Management/CertifyManager/CertifyManager.CertificateRequest.cs

```diff
@@ -729,11 +729,32 @@ private List<string> GetAllRequestedDomains(CertRequestConfig config)
 
                 var challengeConfig = managedCertificate.GetChallengeConfig(domain);
 
+                var credentialsManager = new CredentialsManager();
+
+                Dictionary<string, string> credentials = new Dictionary<string, string>();
+
                 // if our challenge takes a while to propagate, wait
                 if (challengeConfig.ChallengeType == SupportedChallengeTypes.CHALLENGE_TYPE_DNS)
                 {
-                    result.ChallengeResponsePropagationSeconds =
-                        60; // TODO: make cconfig based on max delay required by providers used
+                    if (!String.IsNullOrEmpty(challengeConfig.ChallengeCredentialKey))
+                    {
+                        // decode credentials string array
+                        try
+                        {
+                            credentials = await credentialsManager.GetUnlockedCredentialsDictionary(challengeConfig.ChallengeCredentialKey);
+                        }
```

Comment:
Let's remove these changes to store/get the propagation delay here and in the provider credentials, we'll make it a default on the provider instead. Eventually want to find max of all providers users in the request and wait based on that.